pkg/upgrade/inplace/inplace_test.go

```diff
@@ -0,0 +1,209 @@
+package inplace_test
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	bootstrapv1 "github.com/canonical/cluster-api-k8s/bootstrap/api/v1beta2"
+	controlplanev1 "github.com/canonical/cluster-api-k8s/controlplane/api/v1beta2"
+	"github.com/canonical/cluster-api-k8s/pkg/upgrade/inplace"
+)
+
+func TestIsUpgraded(t *testing.T) {
+	g := NewWithT(t)
+
+	for _, tc := range []struct {
+		name          string
+		annotations   map[string]string
+		releaseString string
+		isUpgraded    bool
+	}{
+		{
+			name:          "notUpgraded",
+			annotations:   map[string]string{bootstrapv1.InPlaceUpgradeReleaseAnnotation: "v1.30"},
```

Comment:
Let's change the v1.30 and v1.29 here, not that it's important in passing the test but only that it makes more sense because we expected the machine actual version be lower than the upgrade goal.